frontend/src/scenes/settings/user/PersonalAPIKeys.tsx

```diff
@@ -298,13 +299,25 @@ function PersonalAPIKeysTable(): JSX.Element {
                             const orgNames = restrictedOrgs.map((org: any) => org.name)
                             const tooltipMessage =
                                 orgNames.length === 1
-                                    ? `Organization <strong>${orgNames[0]}</strong> has restricted the use of personal API keys.`
-                                    : `Organizations <strong>${orgNames.join(
-                                          ', '
+                                    ? `Organization <strong>${escapeHtml(
+                                          orgNames[0]
+                                      )}</strong> has restricted the use of personal API keys.`
+                                    : `Organizations <strong>${escapeHtml(
+                                          orgNames.join(', ')
                                       )}</strong> have restricted the use of personal API keys.`
 
                             return (
-                                <Tooltip title={<span dangerouslySetInnerHTML={{ __html: tooltipMessage }} />}>
+                                <Tooltip
+                                    title={
+                                        <span
+                                            dangerouslySetInnerHTML={{
+                                                __html: DOMPurify.sanitize(tooltipMessage, {
+                                                    ALLOWED_TAGS: ['strong'],
+                                                }),
```

Comment:
`DOMPurify` is all that's strictly needed to fix this. However, it results in removing values from the org name entirely. For example, `<img src='#' onerror=alert(1) />` becomes a blank string. This is probably fine, since legitimate users are unlikely to be using a name like this, but we can make it nicer. `escapeHTML` is thus introduced purely to ensure the full original value is able to be displayed.

Note that either solution alone would fix this XSS. Using `escapeHTML` when we inject the org/project name, without using DOMPurify at all, would prevent the XSS. It would however be potentially error prone and make it easier to reintroduce this in the future if we changed the tooltip message.